### DISCUSSION TITLE
How to get operation space dynamic?

### DISCUSSION THREAD

**Discussion by skywoodsz** [stack-of-tasks/pinocchio#1934] [Q&A]
Hi, I want to design an impendance controller. Therefore, I want to get operation space dynamic, e.g. cartesian inertia matrx, cartesian coriolis and centrifugal matrix. How can I get them using pinocchio? Thanks!

**Comment by jcarpent** [comment]
You can easily use all the routines and algorithms provided by Pinocchio to write down your operational space dynamics.

**Reply by skywoodsz** [reply]
Does this mean that Pinocchio does not provide related functions?
Need customers to write their own program calculation?

**Reply by jcarpent** [reply]
We could think of adding some basic features in the near future, but currently, we have chosen to perform task-space control within an external library such as [TSID](https://github.com/stack-of-tasks/tsid). This choice is to simplify the deployment as Pinocchio end-users might not need such formulation, which often requires an external QP solver such [ProxQP](https://github.com/Simple-Robotics/proxsuite) to solve the task.
What are your needs @skywoodsz?
